crates/vm2/src/tracing.rs

```diff
@@ -335,11 +335,10 @@ impl<T, W> CallframeWrapper<'_, T, W> {
         if self.frame.near_calls.is_empty() || self.near_call == Some(0) {
```

Comment:
I admit I haven't thoroughly looked at this piece of code before, so: how is `near_call` defined? In the `current_frame()` impl, it seems to align with `frame.near_calls` indexing, but in `callframe()`, the opposite indexing is used; e.g., `callframe(0)` will translate to `near_call == None` or `Some(0)`, depending on whether there are near calls in the frame. AFAIU, the logic below and `exception_handler` methods assume reverse indexing. IMO, it would be easier to wrap a hand around direct indexing; in this case, the near call on top can be retrieved using `self.frame.near_calls.get(next_index)`, where `next_index = self.near_call.map_or(0, |idx| idx + 1)`.

Reply:
Yes, it is probably better to do the index reversal earlier.
Also, even with this fix CallTracer doesn't behave correctly, so I think I need to check for other issues, too.